File: Kode eksempler/loaders.cpp
#include <opencv2/opencv.hpp>

using namespace cv;

std::string imagePath = "macaw.jpg";

Mat loadImage(std::string imagepath, std::string colorSpace) {
    Mat result;
    Mat image = imread(imagePath);
    if (colorSpace == "gray") {
        cvtColor(image, result, COLOR_BGR2GRAY);
    } 
    else if (colorSpace == "rgb") {
        cvtColor(image, result, COLOR_BGR2RGB);
    }
    else if (colorSpace == "hsv") {
        cvtColor(image, result, COLOR_BGR2HSV);
    }
    else if (colorSpace == "bgr") {
        result = image;
    }
    return result;
}

void averageIntensities(Mat image) {
    
    cv::Scalar mean = cv::mean(image);
    // Print the average pixel values for each channel (BGR format)
    std::cout << "Average Blue Channel Value: " << mean[0] << std::endl;
    std::cout << "Average Green Channel Value: " << mean[1] << std::endl;
    std::cout << "Average Red Channel Value: " << mean[2] << std::endl;
}

int main() {
    Mat grayImage = loadImage(imagePath, "gray");
    Mat hsvImage = loadImage(imagePath, "hsv");
    Mat rgbImage = loadImage(imagePath, "rgb");
    Mat bgrImage = loadImage(imagePath, "bgr");
    //imshow("Gray", grayImage);
    //imshow("BGR", bgrImage);
    //waitKey();
    averageIntensities(bgrImage);
    return 0;
}

File: Kode eksempler/loaders.py
import cv2 as cv

imagePath = "CPP/Computer Vision/2025 January/macaw.jpg"

def loadImage(imagePath, colorSpace):
    img = cv.imread(imagePath) # Load image
    if colorSpace == "gray":
        result = cv.cvtColor(img, cv.COLOR_BGR2GRAY) # Convert to grayscale
    elif colorSpace == "rgb":
        result = cv.cvtColor(img, cv.COLOR_BGR2RGB)
    elif colorSpace == "hsv":
        result = cv.cvtColor(img, cv.COLOR_BGR2HSV)
    else:
        print(f"Colorspace {colorSpace} is not configured.")
    return result


def main():
    grayImg = loadImage(imagePath, "gray") 
    rgbImg = loadImage(imagePath, "rgb")
    hsvImg = loadImage(imagePath, "hsv")
    bgrImg = cv.imread(imagePath)
    cv.imshow('Grayscale', grayImg) # Display the grayscale image
    cv.imshow('RGB', rgbImg)
    cv.imshow('BGR', bgrImg)
    cv.waitKey(0)
    cv.destroyAllWindows()

main()